### ISSUE TITLE
META_TYPES[DDL_TRIGGER] doesn't exist in old FB servers

### ISSUE THREAD

**Issue by madmak2005** [red-soft-ru/redexpert#11]
There is a problem in SQLUtils.java. Function **generateNameForDBObject** assume all FB databases have DDL_TRIGGER.
```
if (Objects.equals(type, META_TYPES[TRIGGER])) {
            databaseObjects.addAll(databaseHost.getDatabaseObjectsForMetaTag(META_TYPES[DATABASE_TRIGGER]));
            databaseObjects.addAll(databaseHost.getDatabaseObjectsForMetaTag(META_TYPES[DDL_TRIGGER]));
}
```
For old firebirds (I believe <3) this gives null:
`databaseHost.getDatabaseObjectsForMetaTag(META_TYPES[DDL_TRIGGER])`

**Comment by romansimakov** [comment]
What do you expect to get in this case? FB servers before 3 don't have DDL triggers.

**Comment by madmak2005** [comment]
For now I just did as you can see below. Original code gives me a null
pointer exception when using FB 2.5 and trying to just open an *table
trigger*.

if (Objects.equals(type, META_TYPES[TRIGGER])) {
            List<NamedObject> dbo =
databaseHost.getDatabaseObjectsForMetaTag(META_TYPES[DATABASE_TRIGGER]);
            if (dbo != null)
                databaseObjects.addAll(dbo);
            dbo =
databaseHost.getDatabaseObjectsForMetaTag(META_TYPES[DATABASE_TRIGGER]);
            if (dbo != null)
                databaseObjects.addAll(dbo);
}

pt., 9 sie 2024 o 15:21 Roman Simakov ***@***.***> napisał(a):

**Comment by romansimakov** [comment]
Ah, you mean Null Pointer Exception when said null. Ok. It's definitely a bug. Thanks.

**Comment by madmak2005** [comment]
sorry, there was a copy-past error. I mean:

if (Objects.equals(type, META_TYPES[TRIGGER])) {
            List<NamedObject> dbo =
databaseHost.getDatabaseObjectsForMetaTag(META_TYPES[DATABASE_TRIGGER]);
            if (dbo != null)
                databaseObjects.addAll(dbo);
            dbo =
databaseHost.getDatabaseObjectsForMetaTag(META_TYPES[DDL_TRIGGER]);
            if (dbo != null)
                databaseObjects.addAll(dbo);
}

pt., 9 sie 2024 o 15:53 Tomasz Makowski ***@***.***> napisał(a):

**Comment by mikhan808** [comment]
fixed in 2024.09
